### ISSUE TITLE
bloch sphere visualization

### ISSUE THREAD

**Issue by anandijain** [QuantumBFS/Yao.jl#255]
This might be outside the scope of the package, but I am learning quantum computing and using the symbolic `@bra_str` and `@ket_str` to do some basic operations.

I think it would be really cool to visualize individual qubits on the bloch sphere.

please close if this is not relevant.

thanks

**Comment by GiggleLiu** [comment]
Yeah, it should be cool. We just do not have a proper visualization tool for it yet.
We will try to implement one in https://github.com/QuantumBFS/YaoPlots.jl in the future.

**Comment by Roger-luo** [comment]
Hi, I don't think this will be in YaoPlots anytime soon since YaoPlots will focus on more serious plottings targetting research and papers. visualizing individual qubits does not have a concrete use case so far if you just want to learn about the behavior of the gates, this is fairly easy to implement using [Makie.jl](https://github.com/JuliaPlots/Makie.jl) this will even allow animations not just 3D plot. you can find it here: https://github.com/QuantumBFS/SSSS/blob/master/4_quantum/bloch_sphere.jl

The script is only compatible with Yao's previous release v0.5. But feel free to update and submit a PR to YaoPlot. Updating it to v0.6 should be fairly easy.

**Comment by GiggleLiu** [comment]
Can we move this issue to YaoPlots? It should be easy to implement with the Luxor backend.
We can make it a bounty issue @Roger-luo .

**Comment by GiggleLiu** [comment]
Already available in YaoPlots:

```julia
julia> rho = density_matrix(rand_state(2), 1)
DensityMatrix{2, ComplexF64, Array...}
    active qubits: 1/1
    nlevel: 2

julia> bloch_sphere("ρ"=>rho)
```

<img width="467" height="391" alt="Image" src="https://github.com/user-attachments/assets/208d7e4d-8896-42c3-8a62-4ce26193b188" />
